### PR TITLE
Signal traffic improved

### PR DESCRIPTION
**Pull request by aardschok**
- Improved and corrected when signal need to emit if an option is changed.
- Removed the signaling when the camera is changed due it not being stored in the presets.
- Added more signaling to the display and viewport widget
- Colorpicker has an "valueChanged" signal

### CODE REVIEW

File: capture_gui/plugins/cameraplugin.py [hunk 1]
```diff
@@ -37,10 +37,7 @@ def __init__(self, parent=None):
         # Signals
         self.get_active.clicked.connect(self.set_active_cam)
         self.refresh.clicked.connect(self.on_refresh)
-
-        self.cameras.currentIndexChanged.connect(self.options_changed)
-        self.options_changed.connect(self.on_update_label)
-        self.options_changed.connect(self.validate)
```
Comment (BigRoy) [inline capture_gui/plugins/cameraplugin.py]: Doesn't the camera still need to be "validated" when switching?

File: capture_gui/plugins/codecplugin.py [hunk 2]
```diff
@@ -48,6 +48,11 @@ def __init__(self, parent=None):
             if index != -1:
                 self.compression.setCurrentIndex(index)
 
+    def connections(self):
```
Comment (BigRoy) [inline capture_gui/plugins/codecplugin.py]: I think you're missing a function call to this?

File: capture_gui/plugins/cameraplugin.py (hunk 1 above)
Comment (aardschok) [inline capture_gui/plugins/cameraplugin.py]: Reconnecting :
`self.cameras.currentIndexChanged.connect(self.validate)`

**Comment by BigRoy** [comment]
Code looks good; @aardschok can you confirm all the features are now correctly triggering state changes in the interface?

**Comment by aardschok** [comment]
Found an issue with displayGradient which is not passing through for a reason, investigating now.

**Comment by aardschok** [comment]
pre-stated issues resolved

File: capture_gui/app.py [hunk 3]
```diff
@@ -81,6 +81,7 @@ def refresh(self):
 
         with lib.no_undo():
             options = self.options_getter()
+            print options["display_options"]
```
Comment (BigRoy) [inline capture_gui/app.py]: Remove debugging print statement. :)

File: capture_gui/plugins/displayplugin.py [hunk 4]
```diff
@@ -128,7 +128,11 @@ def display_gradient(self):
         define the gradient otherwise the background color will be used as 
         solid color.
         """
-        return self.display_type.currentText() == "Gradient"
+        state = True
```
Comment (BigRoy) [inline capture_gui/plugins/displayplugin.py]: Why did this need changing? The old implementation seemed more readable?

Comment (aardschok) [inline capture_gui/plugins/displayplugin.py]: I am not 100% sure but I think it is a left over of figuring out why the gradient setting got overwritten.
Less code is less errors, I will revert.

**Comment by BigRoy** [comment]
Thanks @aardschok!

**Pull request merged**